core/perf_map.ml

```diff
@@ -0,0 +1,170 @@
+open! Core
+open! Async
+
+type t = Perf_map_location.t Int64.Map.t
+
+let parse_line line =
+  (* empty string for the last line in a file *)
+  if String.is_empty line
+  then None
+  else (
+    try
+      Scanf.sscanf line "%Lx %x %s%s" (fun start_addr size function_ rest ->
+          if String.is_empty rest
+          then
+            ( start_addr
+            , { Perf_map_location.start_addr
+              ; size
+              ; function_
+              ; module_ = None
+              ; file = None
+              ; line = None
+              } )
+          else
+            Scanf.sscanf rest "%s@:%s@:%d" (fun module_ file line ->
+                ( start_addr
+                , { Perf_map_location.start_addr
+                  ; size
+                  ; function_
+                  ; module_ = Some module_
+                  ; file = Some file
+                  ; line = Some line
+                  } )))
+      |> Some
+    with
+    | exn ->
+      raise_s
+        [%message
+          "error scanning perf map, please add this line to a test in perf_map.ml"
+            (exn : exn)
+            (line : string)])
+;;
+
+let%test_module _ =
+  (module struct
+    open Core
+
+    let%expect_test "example perf map" =
+      let sample =
+        {|3a6caa49c2e f3 LazyCompile:~afterInspector node:internal/errors:753
+3a6caa49e86 1e LazyCompile:~lazyInternalUtilInspect node:internal/errors:185
+3a6caa4a5fe 5 Eval:~ node:internal/tty:1
+3a6caa4aa5e 166 Function:~ node:internal/tty:1
+3a6caa4b536 3f LazyCompile:~hasColors node:internal/tty:222
+3a6caa4c3ae 2fa LazyCompile:~getColorDepth node:internal/tty:106
+3a6caa4c90e 5 LazyCompile:~get node:internal/util/inspect:352
+3a6caa5142e 18b LazyCompile:~inspect node:internal/util/inspect:292
+3a6caa51926 1cf LazyCompile:~formatValue node:internal/util/inspect:745
+3a6caa5241e cfd LazyCompile:~formatRaw node:internal/util/inspect:820
+3a6caa537c6 13f LazyCompile:~getConstructorName node:internal/util/inspect:567
+3a6caa53a96 d LazyCompile:~isInstanceof node:internal/util/inspect:559
+|}
+      in
+      String.split sample ~on:'\n'
+      |> List.map ~f:(fun line -> line, parse_line line)
+      |> [%sexp_of: (string * (int64 * Perf_map_location.t) option) list]
```

Comment:
```suggestion
      |> [%sexp_of: (string * (Int64.Hex.t * Perf_map_location.t) option) list]
```

here too I guess, so that it's obvious that e.g. `3a6caa49c2e` is getting parsed correctly as `4014899239982`

Reply:
done